syne_tune/optimizer/schedulers/multiobjective/random_scalarization.py

```diff
@@ -0,0 +1,94 @@
+# Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+from typing import Dict, Optional, Iterable, List, Callable
+
+import numpy as np
+import scipy.stats
+from numpy.random import RandomState
+
+from syne_tune.optimizer.schedulers.searchers.bayesopt.tuning_algorithms.base_classes import (
+    Predictor,
+    ScoringFunction,
+)
+from syne_tune.optimizer.schedulers.searchers.utils.common import Configuration
+
+
+class MultiObjectiveLCBRandomLinearScalarization(ScoringFunction):
+    """
+    Note: This is the multi objective random scalarization scoring function based on the work of Biswajit et al. [1].
+    This scoring function uses Lower Confidence Bound as the acquisition for the scalarized objective
+    :math:`h(\mu, \sigma) = \mu - \kappa * \sigma`
+
+        | [1] Paria, Biswajit, Kirthevasan Kandasamy and Barnabás Póczos.
+        | A Flexible Framework for Multi-Objective Bayesian Optimization using Random Scalarizations.
+        | Conference on Uncertainty in Artificial Intelligence (2018).
+
+    :param predictor: Surrogate predictor for statistics of predictive distribution
+    :param weights_sampler: Callable that can generate weights for each objective.
+        Once called it will return a dictionary mapping metric name to scalarization weight as
+        {
+            <name of metric 1> : <weight for metric 1>,
+            <name of metric 2> : <weight for metric 2>,
+            ...
+        }
+    """
+
+    def __init__(
+        self,
+        predictor: Dict[str, Predictor],
+        active_metric: Optional[List[str]] = None,
+        weights_sampler: Optional[Callable[[], Dict[str, float]]] = None,
+        kappa: float = 0.5,
+    ):
+        self.kappa = kappa
+        self.predictor = predictor
+        if active_metric is None:
+            active_metric = [name for name in predictor.keys()]
+        self.active_metric = active_metric
+
+        if weights_sampler is None:
```

Comment:
No. We want to control all random effects with a seed. I'd do it like this:

Add argument `random_seed: Optional[int] = None`.
This argument is used only if `weights_sampler` is not given. In that case:
`state = RandomState(random_seed)`

I comment above how to assign this random seed.

This is quite important to me. I'd like to retain the property that if a method is re-run with the same seed, then the random sampling behaviour should be exactly the same, as long as this is at all possible.

Reply:
If `random_seed` is `None`, you can still set it to 123 here if you like.

Reply:
An alternative is to provide a default implementation for the `weights_sampler` mapping, as a call with `__call__` method. The class would be generated with `random_seed` and would create `state` as member, and then use it in each `__call__`. This is maybe cleaner.

Reply:
As you prefer, just we'd like to control the seed.